extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/AuthenticationRedirectException.java

```diff
@@ -19,8 +19,12 @@ public AuthenticationRedirectException(int code, String redirectUri) {
         this.redirectUri = redirectUri;
     }
 
+    public AuthenticationRedirectException(Boolean autoRedirect, String redirectUri) {
+        this(autoRedirect ? 302 : 444, redirectUri);
```

Comment:
I'm not sure about using `444`. IMO, we should handle such requests are if they were made to `service` type applications, when using a bearer token.

A `401` response would indicate an invalid session at the application so that the SPA can perform the redirect without necessarily having to parse response headers. As for `403` it would mean the session is valid but access is denied.

From a SPA perspective, after receiving a `401` response, it should be just a matter of calling `window.reload` to start the authentication flow again.

Reply:
Agreed. 444 was just a placeholder. So, it should return a 401, correct? The 403 would already be handled by the general Auth classes would take care of that case, no?

Reply:
@pedroigor @boosey Hi. I think there should be some code followed by a custom header indicating it is to do with the XHR. Otherwise SPA will try to reload even if it is a genuine `401` (whenever we decide to return `401`, example, we return `401` in case the code flow fails to complete). 
To be honest I'm not sure it should be `401`. At the quarkus OIDC level  we want to redirect, right ? But such that if it is XHR and the option is set, then let XHR fail. Can we set `399`, at least it in the `3xx` redirect space ? 
But in any case, it should be accompanied by something like `XHR-Location: redirect_uri...` IMHO

Reply:
I think that works too, +1. I thought `401` would be simpler due to the fact that this status would also mean a `window.reload` at the SPA, the actual authentication flow is managed by the server-side.

Reply:
Hi Pedro @pedroigor Yeah, hopefully some non standard `3xx` should be enough. `401` would be good too but my concern it can confuse SPA and cause it to loop :-) (401 during the original request, then the redirect happens, something fails with the code flow, we return another 401, etc). We can make the code configurable too going forward for sure